app/api/events.py

```diff
@@ -89,6 +89,36 @@ def get_upcoming_events(request: Request, token: AccessTokenPayload = Depends(op
 
     return [Event.parse_obj(event) for event in upcoming_events]
 
+@router.get('/joined-events')
+def get_joined_events(request: Request, token: AccessTokenPayload = Depends(authorize)):
+    """ Returns all (upcoming) events user has joined """
+    db = get_database(request)
+    member = db.members.find_one({'id': UUID(token.user_id)})
+    if not member:
+        raise HTTPException(404, "User could not be found")
+
+    now = datetime.now()
+
+    try:
+        date_str = now.strftime("%Y-%m-%d %H:%M:%S")
+        date = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S")
+    except ValueError:
+        raise HTTPException(500, "Problem handling date format")
+    
+
+    search_filter = {'date': {"$gt": date}}
+
+    # Events to be returned
+    pipeline = [
```

Comment:
I'm not sure we need to use aggregate here. I think something like this maybe could work:

```python
db.events.find({ '$and': { 'date': { '$gt': date }}, 'participants': { '$elemMatch': { 'id': member['id'] } }})
```

(I am not sure though, so maybe it is needed😅)

Reply:
Didn't work, might have to keep aggregate for now

Reply:
The reason pipeline is used here is that ElemMatch returns the participant object and not the entire event

Reply:
Okay my bad then😅